### PR TITLE
Set interstitial last seen to future date when clearing school_info_id in users

### PR DESCRIPTION
**Pull request by jopolsky**
Setting the interstitial dialog to show in the future.

Related to: https://github.com/code-dot-org/code-dot-org/pull/18254

### CODE REVIEW

File: bin/oneoff/data_fix/fix_bad_school_info_in_users [hunk 1]
```diff
@@ -17,15 +17,26 @@
...
+  end
+  opts.on('-d', '--interstitial-days DAYS', 'Number of days before showing the interstitial dialog.') do |d|
+    options[:interstitial_days] = Integer(d)
+  end
+  opts.on('-h', '--help', 'Prints this help') do
+    puts opts
+    exit
+  end
+end.parse!
+puts "Called with options: #{options}"
```
Comment (aoby) [inline bin/oneoff/data_fix/fix_bad_school_info_in_users:39]: Nice

File: bin/oneoff/data_fix/fix_bad_school_info_in_users [hunk 2]
```diff
@@ -64,8 +75,10 @@ CSV.open(csv_file, 'w') do |csv|
 
     users.each do |user|
       csv << [user.id.to_s, user.school_info_id.to_s]
-      if ACTUALLY_UPDATE
-        user.update!(school_info_id: nil)
+      if options[:actually_update]
+        user.last_seen_school_info_interstitial = DateTime.now + options[:interstitial_days]
+        user.school_info_id = nil
+        user.save!(validate: false)
```
Comment (aoby) [inline bin/oneoff/data_fix/fix_bad_school_info_in_users]: Why skip validation? I'm guessing that's for the same reason as [this](https://github.com/code-dot-org/code-dot-org/blob/8d961a3524e323e867e2d42e31a16ced3a8f4691/lib/cdo/school_info_interstitial_helper.rb#L22-L25), in which case please add a small comment. 

As an aside, we really should try to fix our users table so they all pass validation at some point :(

File: bin/oneoff/data_fix/fix_bad_school_info_in_users [hunk 3]
```diff
@@ -64,8 +75,10 @@ CSV.open(csv_file, 'w') do |csv|
 
     users.each do |user|
       csv << [user.id.to_s, user.school_info_id.to_s]
-      if ACTUALLY_UPDATE
-        user.update!(school_info_id: nil)
+      if options[:actually_update]
+        user.last_seen_school_info_interstitial = DateTime.now + options[:interstitial_days]
```
Comment (aoby) [inline bin/oneoff/data_fix/fix_bad_school_info_in_users:79]: Why are we setting it to today (or the future)? Is this just to give them at least a week before [asking again](https://github.com/code-dot-org/code-dot-org/blob/8d961a3524e323e867e2d42e31a16ced3a8f4691/lib/cdo/school_info_interstitial_helper.rb#L16)?

Comment (jopolsky) [reply bin/oneoff/data_fix/fix_bad_school_info_in_users:79]: Marina asked that we delay the popup until after the school lookup is added. However, they need to fix the data now in order to run the other script that sets the school_info_id based on past PD enrollments.

Comment (aoby) [reply bin/oneoff/data_fix/fix_bad_school_info_in_users:79]: Gotcha. We will (obviously) need to make sure we add the school lookup before 7 days from whenever we set these to.

Comment (jopolsky) [reply bin/oneoff/data_fix/fix_bad_school_info_in_users:79]: We are going to future date it, 35 days from now on November 15th. With the future date the number of days is negative and the check will return false, which means don't show.

**Pull request merged**